unittests/wasm_tests.cpp

```diff
@@ -1843,6 +1845,39 @@ BOOST_FIXTURE_TEST_CASE( depth_tests, TESTER ) try {
 
 } FC_LOG_AND_RETHROW()
 
+// TODO: Update to use eos-vm once merged
+BOOST_FIXTURE_TEST_CASE( code_size, TESTER )  try {
+   using namespace IR;
+   using namespace Runtime;
+   using namespace Serialization;
+   std::vector<U8> code_start = {
+      0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00, 0x01, 0x07, 0x01, 0x60,
+      0x03, 0x7e, 0x7e, 0x7e, 0x00, 0x03, 0x02, 0x01, 0x00, 0x07, 0x09, 0x01,
+      0x05, 0x61, 0x70, 0x70, 0x6c, 0x79, 0x00, 0x00, 0x0a, 0x8b, 0x80, 0x80,
+      0x0a, 0x01, 0x86, 0x80, 0x80, 0x0a, 0x00
+   };
+
+   std::vector<U8> function_body = {
+      0x41, 0xff, 0xff, 0xff, 0xff, 0x07, 0x1a
+   };
+
+   std::vector<U8> code_end = { 0x0b };
+ 
+   std::vector<U8> code_function_body;
+   for(unsigned int i = 0; i < wasm_constraints::maximum_code_size; i+=7) {
+      code_function_body.insert(code_function_body.end(), function_body.begin(), function_body.end());
+   }
+
+   std::vector<U8> code;
+   code.insert(code.end(), code_start.begin(), code_start.end());
+   code.insert(code.end(), code_function_body.begin(), code_function_body.end());
+   code.insert(code.end(), code_end.begin(), code_end.end());
+
+   Module module;
+   Serialization::MemoryInputStream stream((const U8*)code.data(), code.size());
+   BOOST_CHECK_THROW(WASM::serialize(stream, module), FatalSerializationException);
```

Comment:
I'm not really fond of testing internal components that are not directly visible to users.  We don't really care about exactly where the restriction is enforced as long as the user-visible behavior is fixed.